### ISSUE TITLE
taper_cross_section_linear does not taper cladding

### ISSUE THREAD

**Issue by gdspaul** [gdsfactory/gdsfactory#1081]
**Describe the bug**
New taper function doesn't handle cladding regions.  Old `taper2` could do this.

**To Reproduce**
```
xc_sin = partial(
    cross_section,
    width=1.0,
    layer=(0,1),
    cladding_layers=[(0,2), (0,3)],
    cladding_offsets=[5, 10]
)

xc_sin_ec = partial(
    xc_sin,
    width=0.2
)


@gf.cell
def demo():

    taper_length = 10

    in_stub_length = 10
    out_stub_length = 10

    c = gf.Component()

    wg_in = c << gf.components.straight(
        length=in_stub_length,
        cross_section=xc_sin_ec
    )

    taper = c << gf.components.taper_cross_section_linear(
        length=taper_length,
        cross_section1 = xc_sin_ec,
        cross_section2 = xc_sin
    )

    wg_out = c << gf.components.straight(
        length=out_stub_length,
        cross_section=xc_sin
    )

    taper.connect("o1", wg_in.ports["o2"])
    wg_out.connect("o1", taper.ports["o2"])

    c.add_port("o1", port=wg_in.ports["o1"])
    c.add_port("o2", port=wg_out.ports["o2"])

    return c
```
![image](https://user-images.githubusercontent.com/121980307/210652074-82d66487-b9bb-470c-b8e3-1e3f18f52f5b.png)

**Expected behavior**
There should be cladding in the taper region.

**Comment by joamatab** [comment]
Thank you Paul for reporting this, 

in principle only named Sections were supposed to be tapered, but I understand the convenience of defining them as you did in your code


![image](https://user-images.githubusercontent.com/4514346/210660847-1eb52649-1690-4ef6-b841-5034921cada7.png)

**Comment by gdspaul** [comment]
Hi Joaquin, thanks for the quick fix.  (This is Paul Hansen btw, hi!)  Should I do something else to get my tapered cladding?  I'm just learning your cool tool.

**Comment by joamatab** [comment]
Hi Paul,

yes, I was imagining it was you :) thank you for your issue

I just released `6.18.1` which should work as you expect
i think it makes sense to be able to easily define cladding_layers and cladding_offsets and automatically taper them to the tapers

@simbilod  was also asking for this as well

**Comment by gdspaul** [comment]
hey Joaquin, that screenshot you sent above doesn't look right.  i get something similar in my code.  why doesn't the taper section's cladding match the input and output waveguides' cladding?
![image](https://user-images.githubusercontent.com/121980307/210668780-da0a5554-664e-41e1-800f-18f20c3db914.png)
